src/test/java/com/yelp/nrtsearch/server/grpc/QueryTest.java

```diff
@@ -911,6 +922,22 @@ private void testWithBoost(Query originalQuery, SearchResponse searchResponse) {
     }
   }
 
+  private void testWithExplain(Query query, SearchResponse searchResponse) {
+    boolean explain = true;
+    SearchResponse searchResponseExplained =
+        grpcServer.getBlockingStub().search(buildSearchRequestWithExplain(query, explain));
+    for (int i = 0; i < searchResponse.getHitsCount(); i++) {
+      SearchResponse.Hit hit = searchResponse.getHits(i);
+      SearchResponse.Hit explainedHit = searchResponseExplained.getHits(i);
+      SearchResponse.Hit hitWithoutExplain =
+          SearchResponse.Hit.newBuilder(hit).setExplain("").build();
```

Comment:
Nit: `hit.toBuilder()` would be a bit more concise as compared to `SearchResponse.Hit.newBuilder(hit)`. Also if you compare the explanations first you won't need to create a builder to compare fields other than explain.